macos/FluentUI/Button/Button.swift

```diff
@@ -254,6 +254,39 @@ open class Button: NSButton {
 		path.fill()
 	}
 
+	public override func viewDidMoveToWindow() {
+		super.viewDidMoveToWindow()
+		isWindowInactive = !(window?.isMainWindow ?? false)
+
+		if window != nil {
+			// Hook in Notification Handles to capture the Window's active and inactive states
+			resignMainWindowObserver = NotificationCenter.default.addObserver(forName: NSWindow.didResignMainNotification,
+												   object: window,
+												   queue: nil) {[weak self] _ in
+				self?.isWindowInactive = true
+			}
+
+			becomeMainWindowObserver = NotificationCenter.default.addObserver(forName: NSWindow.didBecomeMainNotification,
+												   object: window,
+												   queue: nil) {[weak self] _ in
+				self?.isWindowInactive = false
+			}
+		} else {
+			// If there's no valid window we're likely here because the Button's view moved away from a Window which it was
+			// previously loaded into. Good opportunity to remove the Notification Observers here.
+			if let resignMainWindowObserver = resignMainWindowObserver {
+				NotificationCenter.default.removeObserver(resignMainWindowObserver)
+			}
+			resignMainWindowObserver = nil
+
+			if let becomeMainWindowObserver = becomeMainWindowObserver {
+				NotificationCenter.default.removeObserver(becomeMainWindowObserver)
+			}
+			becomeMainWindowObserver = nil
```

Comment:
We also need to remove observer from the old window if we're moving directly into a new window? So maybe the structure of viewDidMoveToWindow is:
if let observer1 {
removeObserver
}

if let observer2 {
removeObserver
}

if window != nil {
addObserver
}